src/coreComponents/mesh/generators/VTKMeshGenerator.cpp

```diff
@@ -0,0 +1,841 @@
+    {
+      cellToVertex[cellCount][v] = currentCell->GetPointId(v);
+    }
+    localToGlobal[cellCount] = globalCellIdsDataArray.GetValue(c);
+    cellCount++;
+  }
+}
+
+std::unordered_set< string > getMaterialWrapperNames( ElementSubRegionBase const & subRegion )
+{
+  using namespace constitutive;
+  std::unordered_set< string > materialWrapperNames;
+  subRegion.getConstitutiveModels().forSubGroups< ConstitutiveBase >( [&]( ConstitutiveBase const & material )
+  {
+    material.forWrappers( [&]( WrapperBase const & wrapper )
+    {
+      if( wrapper.sizedFromParent() )
+      {
+        materialWrapperNames.insert( ConstitutiveBase::makeFieldName( material.getName(), wrapper.getName() ) );
+      }
+    } );
+  } );
+  return materialWrapperNames;
+}
+
+void importFieldOnCellBlock( string const & name, std::vector<vtkIdType > const & cellIds, int region_id,
+                             ElementRegionManager & elemManager,
+                             std::vector< vtkDataArray * > const & arraysTobeImported)
+{
+  assert(region_id >= -1 );
+  string const cellBlockName = region_id != -1  ? std::to_string(region_id) + "_" + name : name;
+  elemManager.forElementSubRegions< CellElementSubRegion >( [&]( CellElementSubRegion & subRegion )
+  {
+    if( subRegion.getName() == cellBlockName)
+    {
+      /// Writing properties
+      std::unordered_set< string > const materialWrapperNames = getMaterialWrapperNames( subRegion );
+      for( vtkDataArray * vtkArray : arraysTobeImported )
+      {
+        WrapperBase & wrapper = subRegion.getWrapperBase( vtkArray->GetName() );
+        if( materialWrapperNames.count( vtkArray->GetName() ) > 0 && wrapper.numArrayDims() > 1 )
+        {
+          // Scalar material fields are stored as 2D arrays, vector/tensor are 3D
+          GEOSX_LOG_RANK_0("going in with : " << vtkArray->GetName());
+          using ImportTypes = types::ArrayTypes< types::RealTypes, types::DimsRange< 2, 3 > >;
+          types::dispatch( ImportTypes{}, wrapper.getTypeId(), true, [&]( auto array )
+          {
+            using ArrayType = decltype( array );
+            Wrapper< ArrayType > & wrapperT = Wrapper< ArrayType >::cast( wrapper );
+            auto const view = wrapperT.reference().toView();
+
+            localIndex cellCount = 0;
+            for( vtkIdType c : cellIds )
+            {
+              int comp = 0;
+              for( localIndex q = 0; q < view.size( 1 ); ++q )
+              {
+                // The same value is copied for all quadrature points.
+                LvArray::forValuesInSlice( view[cellCount][q], [&]( auto & val )
+                {
+                  val = vtkArray->GetComponent(c,comp);
+                } );
+              }
+              cellCount++;
+              comp++;
+            }
+          });
+        }
+        else
+        {
+          using ImportTypes = types::ArrayTypes< types::RealTypes, types::DimsRange< 1, 2 > >;
```

Comment:
I would wrap what's inside the `if` and the `else` branches in two separate functions a bit as it was done in the `PamelaMeshGenerator`.